e2e/blockstorage/resource_block_storage.go

```diff
@@ -346,15 +368,20 @@ func waitForDetach(apiClient *client.Client, blockStorageID string, project_id i
 		if data["status"] == "Available" {
 			break
 		}
-		// Wait for 2 seconds before checking the status again (is Volume Detached?)
-		time.Sleep(2 * time.Second)
+		// Wait for some time before checking the status again (is Volume Detached?)
+		time.Sleep(constants.WAIT_TIMEOUT * time.Second)
 	}
 	return nil
 }
 
 func validateSize(d *schema.ResourceData, m interface{}) error {
 	apiClient := m.(*client.Client)
 
+	_, err := apiClient.GetAllProjects(d.Get("project_id").(int), d.Get("location").(string))
+	if err != nil {
+		return err
+	}
+
```

Comment:
remvoe this api call. No need to make extra api call

Reply:
done